empress/recon_vis/recon_viewer.py

```diff
@@ -96,83 +222,358 @@ def render_parasite_helper(fig, node, recon, host_lookup, show_internal_labels,
+    if host_node.left_node.name == left_host_node.name:
+        render_curved_line_to(node_xy, left_xy, fig)
+        host_node.layout.lower_v_track += (host_node.layout.x - node_xy[0]) / offset
+    else:
+        stop_row = host_node.left_node.layout.row
+        connect_child_to_parent(node, left_node, host_lookup, recon, fig, stop_row=stop_row)
+
+    #Draw Right node
+    if host_node.right_node.name == right_host_node.name:
+        render_curved_line_to(node_xy, right_xy, fig)
+        host_node.layout.upper_v_track += (host_node.layout.x - node_xy[0]) / offset
+    else:
+        stop_row = host_node.right_node.layout.row
+        connect_child_to_parent(node, right_node, host_lookup, recon, fig, stop_row=stop_row)
+
+
+def get_children(node, recon, parasite_lookup):
+    """
+    Gets the children of a node in the order they appear in the mapping node.
+    :param node: Node object
+    :param recon: Reconciliation Object
+    :param parasite_lookup: Dictionary with parasite node names as the key and parasite node objects as the values
+    :return A tuple consisting of the left node and right node
+    """
+    mapping_node = recon.mapping_of(node.name)
+    event = recon.event_of(mapping_node)
+    left_mapping_node = event.left
+    right_mapping_node = event.right
+    left_node_name = left_mapping_node.parasite
+    right_node_name = right_mapping_node.parasite
+
+    left_node = parasite_lookup[left_node_name]
+    right_node = parasite_lookup[right_node_name]
+
+    return left_node, right_node
+
+def render_curved_line_to(node_xy, other_xy, fig):
+    """
+    Renders a curved line from one point to another
+    :param node_xy: x and y position of a node
+    :param other_xy: x and y position of another node
+    :param fig: Figure object that visualizes trees using MatplotLib
+    """
+    mid_xy = (node_xy[0], other_xy[1])
+    fig.line(node_xy, mid_xy, PARASITE_EDGE_COLOR)
+    fig.line(mid_xy, other_xy, PARASITE_EDGE_COLOR)
+
+def render_transfer_branch(node_xy, right_xy, fig, node, host_lookup, recon, right_node):
+    """
+    Renders a transfer branch
+    :param node_xy: x and y position of a node
+    :param right_xy: x and y position of the right child of a node
+    :param fig: Figure object that visualizes trees using MatplotLib
+    :param node: Node object
+    :param host_lookup: Dictionary with host node names as the key and host node objects as the values
+    :param recon: Reconciliation object
+    :param right_node: The right node object of node
+    """
+    mapping_node = recon.mapping_of(node.name)
+    host_node = host_lookup[mapping_node.host]
+
+    child_mapping_node = recon.mapping_of(right_node.name)
+    child_host_node = host_lookup[child_mapping_node.host]
+
+    #check temporal consistency of transfer event
+    if child_host_node.parent_node.layout.col < node.layout.col:
+        #Draw right node, which is transfered
+        mid_xy = (node_xy[0], right_xy[1])          #xy coords of midpoint
```

Comment:
Please don't use `[0]` or `[1]` for accessing variables like this. Either create a class called Point or a named tuple and access it as `.x` and `.y`.

https://docs.python.org/3/library/collections.html#collections.namedtuple

It might not be trivial to create a class or a named tuple. If you have any problems with this, feel free to contact me.

Reply:
Would it be fine to use node_x, node_y = node_xy?

Reply:
There are some instances where you only access one of the variables (x or y). In those cases, you probably don't want to add an extra line just to make the code more clear. I still think it's better to either use a class or a named tuple.

Here's the syntax for the class if you want to create a class:
```python
class Point:
    def __init__(x: int, y: int):
        self.x = x
        self.y = y
```

Here's the syntax for the class if you want to create a class inherited from named tuple:
```python
from typing import NamedTuple
...
class Point(NamedTuple):
    x: int
    y: int
```
You will put this code either at the top of utils file or at the top of tree files (I think putting it at the top of the tree file is more appropriate). If you put it at the top of the tree file, you would refer to the type as `tree.Point`.
```python
point = tree.Point(5, 10)  # x = 5, y = 10
print(point.x)
print(point.y)
```

Reply:
I'm sorry for not asking you this, but is there a reason you want to use `node_x, node_y = node_xy` over a class?

Reply:
I was mostly asking, for the other parts of our code that already made use of unpacking

Reply:
I think we did what we did because 1. It was how Prof Ran did it when he gave use the code, and 2. We thought it would be easier than having a named tuple or a class. However, we can change it if you want us to!

Reply:
Never mind, I see @TrentonJW already changed it.